Makefile

```diff
@@ -1897,7 +1897,9 @@ clipping_iterator_test: $(OBJ_DIR)/db/compaction/clipping_iterator_test.o $(TEST
 
 ribbon_bench: $(OBJ_DIR)/microbench/ribbon_bench.o $(LIBRARY)
 	$(AM_LINK)
-
+	
+cache_reservation_manager_test: $(OBJ_DIR)/cache/cache_reservation_manager_test.o $(TEST_LIBRARY) $(LIBRARY)
```

Comment:
@pdillinger I know we have talked about whether to keep a separate testing binary in https://github.com/facebook/rocksdb/pull/8506#discussion_r668095996. For this new test, we have 9 test cases (which is more than the previous test with only 1). Does it make it worth itself a separate testing binary?